cgrok/grok_match.c

```diff
@@ -52,8 +52,7 @@ int grok_match_walk_next(const grok_match_t *gm,
   }
 
   *namelen = gct->name_len;
-  *name = malloc(*namelen);
-  memcpy(*name, gct->name, *namelen);
+  *name = gct->name;
```

Comment:
The caller of `grok_match_walk_next` in our use case almost immediately makes a copy (because it needs a Go string). Also, `gct->name` leaks because we don't tear down `grok_capture` structs properly. 

We could free the string in Go after we copy it, I'll dig into why they did this in the first place.